### DISCUSSION TITLE
How to display the code as a page when editing online

### DISCUSSION THREAD

**Discussion by Zhengyueewee2e** [outline/outline#4288] [General]
![image](https://user-images.githubusercontent.com/40627010/195801251-3e8dd8ea-5897-4fde-af10-bf957e994a58.png)

**Comment by almereyda** [comment]
Hi @Zhengyueewee2e,

the code block type that you are showing is intended to display verbatim code blocks. The selection of the language is used for syntax highlighting.

Currently there is no option to add custom HTML document. This is also a safe option, since HTML and associated JS will be displayed within the current security context of you Outline's domain, which means an eventually hostile document could access your Outline instance as your user.

From a request loading a page, we can also learn which security policies are in place:

```json
{
	"Response Headers (682 B)": {
		"headers": [
			{
				"name": "content-encoding",
				"value": "gzip"
			},
			{
				"name": "content-security-policy",
				"value": "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' gist.github.com; style-src 'self' 'unsafe-inline' github.githubassets.com; img-src * data: blob:; frame-src * data:; connect-src *"
			},
			{
				"name": "content-type",
				"value": "text/html; charset=utf-8"
			},
			{
				"name": "date",
				"value": "Fri, 14 Oct 2022 18:18:56 GMT"
			},
			{
				"name": "referrer-policy",
				"value": "no-referrer"
			},
			{
				"name": "strict-transport-security",
				"value": "max-age=15552000; includeSubDomains"
			},
			{
				"name": "timing-allow-origin",
				"value": "https://sub.example.com"
			},
			{
				"name": "vary",
				"value": "Accept-Encoding"
			},
			{
				"name": "x-content-type-options",
				"value": "nosniff"
			},
			{
				"name": "x-dns-prefetch-control",
				"value": "on"
			},
			{
				"name": "x-download-options",
				"value": "noopen"
			},
			{
				"name": "X-Firefox-Spdy",
				"value": "h2"
			},
			{
				"name": "x-frame-options",
				"value": "SAMEORIGIN"
			},
			{
				"name": "x-xss-protection",
				"value": "1; mode=block"
			}
		]
	}
}
```

About generic `<iframe />` support to load external HTML sources in a child container, please follow the discussion in

- https://github.com/outline/outline/discussions/2286
